### PR TITLE
[Balance] Buff the Nuclear Missile

### PR DESCRIPTION
**Pull request by Azure3141**
----------------------
**Content (Artwork / Missions / Jobs)**

## Summary

This buffs the Nuclear Missile reward from FW Checkmate. The Nuclear Missile now works exceptionally well as a "fleet killer" weapon, capable of doing a large amount of damage to a group of ships.

Base shield / hull damage was increased from 9000 / 7000 to 24000 / 16000. While this may seem like a lot (it's about a 2.66x increase for shield damage and a ~2.3x increase in hull damage), the effective damage increase is actually much less. This is because the blast radius and trigger radius of the Nuke has been increased to 300 and 270 respectively. 

According to this chart for the `blast radius` attribute, this ratio of blast radius to trigger radius (90%) results in an effective damage of ~50% of the base weapon damage. This means that the Nuclear Missile effectively does 12000 / 8000 damage to a targeted ship.

 In practice, this results in two nukes disabling a Cruiser, and three killing a Carrier. Eight nukes are needed to (barely) disable a KIV. This compares with the current nuke, where three nukes disables a Cruiser, four nukes kills a Carrier, and seventeen nukes are needed to kill a KIV (that one's probably so different because Sestor ships tend to have a lot of hull).

Additionally, the Nuclear Missile will only detonate when its target it inside its blast radius (300 pixels), so a nuke hitting a stray asteroid or rogue enemy drone won't make it vaporize everything in the area.

For reference, a Sidewinder does 1300 / 900 shields / hull damage per ton, while the existing nuke does 900 / 700 shield / hull damage person, which is laughable. This PR brings that up to the more respectable values of 2400 / 1600 shield / hull damage per ton, a bit over the damage / ton of the Typhoon for shields, but still below for hull (which has a damage / ton of 2100 / 2100).

This all comes together to make the Nuclear Missile more viable as a weapon against massed targets. If you're willing to spend eight gunports on something like a Weapons Kestrel, you can even kill a single T2 HW! While I don't think this fully addresses the flaws of Checkmate's rewards, it does at least make the nuke somewhat less of a joke.

### CODE REVIEW

**Comment by Zitchas** [comment]
I like the increased damage, not so much the odd safety mechanism. These are supposed to be indiscriminate weapons of mass destruction that are so terrible everyone agreed to never use them again; not smart munitions with advanced safety mechanisms to ensure they only go off in proximity to their target. I'm ambivalent about the outfit space change, neither liking nor disliking it.

**Comment by Amazinite** [comment]
> The size of the nuke has also been dropped to 8 tons, ~~because Derpy said he wanted to do that anyway.~~

I never said that. :derpy:

**Comment by Saugia** [comment]
I think it would make sense more for the [Avgi] but it makes sense that humans using nukes is rather dangerous. We shouldn't be pushing the narrative that nukes are a good thing to use.

**Comment by Saugia** [comment]
(It seems I mis-understood `safe`. Carry on.)

**Comment by Iris-Persephone** [comment]
(I've been chatting on the Discord for a bit, but I finally decided to make an account. Hi!)

As one of the people who favoured buffing nukes, I feel I should make a few points. 

When nukes start showing up in late FW and later as the Checkmate reward, it should feel like a big deal. Unfortunately, at present, they're rather underwhelming for the player to use, and compared to the alternative from the Reconciliation branch, almost useless. The "big weapon against massed targets" angle is a valid niche - in compensation for the extreme cost and the loss of one of your weapon slots (as well as a not insignificant chance of vaporizing something important), you can deal heavy damage to a group of ships that you may not have been able to damage otherwise.

I don't think that giving someone a reason to use nukes is asserting they are good - rather, it's a choice which comes with significant risks and morality issues, but one that a player may still choose to make. For similar reasons, I think changing the various other rewards of FW so that the nuke looks better in comparison just steps around the real issue. If, when given a choice, people don't use nukes, it doesn't mean that the alternatives should be nerfed to force them to consider nukes - it means that something is wrong with nukes and they aren't seen as "worth it". In that case, I think a buff would be in order.

**Comment by ziproot** [comment]
+1 on everything here. The safety feature should also make it easier to do the nuke dodging in FW reconciliation because if you are cloaked none of the nukes will go off.

**Review by Zitchas — CHANGES_REQUESTED** [review]
I feel that changing the nuke from "explode when near *any* enemy" to "explode only when near the original intended target" is going to change it from being something scary to something that's even less of a concern than a javelin... unless the player is actually the target of it. This sounds like just another push to have escorts, as so long as the nuke carrying ship is focused on literally *anyone* else other than the player, then the player has virtual immunity to the nukes so long as they aren't close to those other ships. 

As far as nukes detonating against asteroids goes, I see that as being more of an alternative safety mechanism. Not safety as in "ensure it doesn't blow up unintended targets", but rather safety as in "ensure that no matter what happens, this nuke will go off." Nukes aren't guns. They're not regular missiles. They are (intended to be) apocalypse weapons of mass destruction. As in, you don't pull the trigger on a nuke until you want someone's world to end, and you're not overly concerned about *your* world anymore. If the nuke launches, it should go off. If we had the mechanics for it, I'd go so far as to say they should be set up so that being shot down by AM actually detonates them on the spot, just with reduced explosive radius and/or damage. Blowing up against an asteroid isn't ideal, but it's better (from the point of view of a deployer of WMD) to have it blow up a little early against an asteroid than to have it not blow up at all. They should also detonate when they reach the end of their range, regardless of proximity to anything. 

In other words, make these things *terrifying*, and a guaranteed explosion. Don't try to make them just a bigger & better torpedo.

**Comment by Zitchas** [comment]
Why does it have to be easy and convenient to use? I don't think that figures anywhere in the intent or purpose of the nuclear missile.

And if the purpose of this is to actually make checkmate's rewards as good as the standard FW path's rewards... Nothing short of giving the player access to a launcher that launches nukes and produces its own ammo will come anywhere close. Probably not even then, honestly. 

At the end of the day, this is a single shot missile. Nothing is going to make it good for the player in the current combat environment. Best we can hope for is to be terrifying. (and, to be fair, I do find it scary. It definitely one-shoted more than a few of my flagships over the years. So unless something in our mechanics has changed to make it weaker than it used to be, I don't even consider it a given that it needs changing at all.) Not every weapon has to be convenient for the player.

**Comment by Quantumshark** [comment]
The first fight against the Pug in FW checkmate is nigh impossible with these nukes.
The Pug arrive slowly, a few ships at a time, and thus only lose a few ships to the Wolfpack nukes (while the wolfpack lose their own ships immediately by nuking themselves and thus only ever fire one shot). The FW fleet takes off all at once, and loses most of its ships immediately if any of the wolfpack ships spawn anywhere near Silver. Even if the FW fleet survives, the Pug vastly outnumber them and the usual end result is at least three surviving Maboros and several smaller ships to no surviving FW ships.

The nukes are also almost impossible to use - the firing ship is very likely to be destroyed if it's within 600 range of the target, the missiles are far more likely to strike an asteroid than the target if fired outside this range, and the huge trigger radius means the target will usually only take a survivable glancing blow from the nuke, which pug ships will rapidly recover from. The trigger radius can be circumvented by targeting the ship at the back of a group, but this is neither intuitive nor particularly convenient to do.

Savefile, if anyone else wants to try it out:
[William Nucleararmament~Checkmate Pug.txt](https://github.com/endless-sky/endless-sky/files/11591799/William.Nucleararmament.Checkmate.Pug.txt)

**Comment by Quantumshark** [comment]
I turned the trigger radius on the nukes down to 30, and after that it's only taken me a couple of days to beat FWC Pug 1 (which is still probably more restarting than it actually should take). This would probably make them too dangerous for the reconciliation side, though, so that probably isn't a good solution. It really does seem that making nukes powerful enough to be useful for the player necessarily implies making them far too powerful when used by NPCs, and I'm really not sure what to do about that.

**Comment by Hurleveur** [comment]
We can just adapt the fleets to take into account the fact the nukes dont do damage to pugs

**Comment by Amazinite** [comment]
A concept I've floated in the past is giving nukes a launcher that can hold multiple warheads instead of having them be one-and-done. I'd prefer we go that route instead of massively buffing the one-and-done warhead outfit. This new outfit would be unlocked at the end of or during the epilogue of FWC, meaning that the nukes you encounter during the main campaign wouldn't be buffed (therefore making the campaign more difficult, as @Quantumshark has highlighted), increase providing the player a stronger weapon as a campaign reward.
